starlite/response/template.py

```diff
@@ -36,16 +39,28 @@ def __init__(
                 [BackgroundTasks][starlite.datastructures.BackgroundTasks] to execute after the response is finished.
                 Defaults to None.
             headers: A string keyed dictionary of response headers. Header keys are insensitive.
-            cookies: A list of [Cookie][starlite.datastructures.Cookie] instances to be set under the response 'Set-Cookie' header.
+            cookies: A list of [Cookie][starlite.datastructures.Cookie] instances to be set under the response
+                'Set-Cookie' header.
             encoding: Content encoding
+            media_type: A string or member of the [MediaType][starlite.enums.MediaType] enum. If not set, try to infer
+                the media type based on the template name. If this fails, fall back to `text/plain`.
         """
+        if media_type == MediaType.JSON:  # we assume this is the default
+            suffixes = PurePath(template_name).suffixes
+            for suffix in suffixes:
```

Comment:
This code block can be polished i think.

Reply:
I tried a few variations but I think this is the cleanest. Do you have any suggestions?

As I said in the issue, we have to assume `MediaType.JSON` is the default, since it will always be set - there's no way to receive `None` here, unless we change how / when we set default media types completely.

As for the suffixes part, `os.splitext` only returns the last suffix, which would break for common naming conventions such as `template.html.jinja2`. I could re-implement `PurePath.suffixes` here, but what would be the point of that?